src/WebSocket/WebSocketDispatcher.ts

```diff
@@ -0,0 +1,114 @@
+/*
+*  Send and receive requests serially
+*  next reuqest is only sent when the previous one got its response
+*/
+
+import { Dispatcher } from '../api';
+
+const errorMessages = {
+  notConnected: 'Not connected!',
+  resourceIsBusy: 'Resource is busy!',
+  unknownState: 'Error due to unknown state',
+};
+
+export class WebSocketDispatcher extends Dispatcher {
+  private webSocket?: WebSocket;
+  private responseData: any;
+
+  open(): Promise<null> {
```

Comment:
why you put null? Don't put anything in resolve and the returning type should be void

Reply:
is open an inner method? in this case it's **_open**

Reply:
open method could be used externally
type null means no value, void is just the same

Reply:
void - means no type, mostly used for functions without any return statement
null - for cases when you want to specify "no-value"
Promise<void>  - seems incorect, Promise<null> - much more meaningfull

Reply:
You were right, Dmitriy. Promise.resolve() is of type Promise<void>, not Promise<null>
Need to re-factor a little.